tools/find_deprecated.py

```diff
@@ -236,6 +249,34 @@ def find_deprecations(file_name: Path) -> list[Deprecation]:
                     DETAILS += f" (wrt last minor release, {round(diff_days / 30.4)} month old)"
             except KeyError:
                 DETAILS = "Future release"
-        print(f"\n{since_version}: {DETAILS}")
+        lines = []
         for deprecation in deprecations:
-            print(f" - {deprecation.location_str} ({deprecation.target})")
+            is_pending = " - PENDING" if deprecation.pending else ""
+            if pending == "include":
+                lines.append(f" - {deprecation.location_str} ({deprecation.target}){is_pending}")
+            if pending == "only" and deprecation.pending:
+                lines.append(f" - {deprecation.location_str} ({deprecation.target}){is_pending}")
+            if pending == "exclude" and not deprecation.pending:
+                lines.append(f" - {deprecation.location_str} ({deprecation.target})")
```

Comment:
You can leverage early returns to make this easier to read.

```suggestion
            if pending == "exclude" and deprecation.pending:
                continue
            if pending == "only" and not deprecation.pending:
                continue
            pending_str = " - PENDING" if deprecation.pending else ""
            lines.append(f" - {deprecation.location_str} ({deprecation.target}){is_pending}")
```

Also consider renaming the variable to `pending_arg` to make it explicit what it is.

Reply:
`pending_arg` renaming in 8145a6332560ccb07b7ac888ae68652d6a41088a